src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -308,6 +315,10 @@ def process_public_ip_create_namespace(namespace):
     if namespace.dns_name:
         namespace.public_ip_address_type = 'dns'
 
+def load_cert_file(namespace):
+    if os.path.isfile(namespace.public_cert_data):
```

Comment:
this supports having raw data passed in on the command line, so we leave the data alone if it is not a file 

---
In reply to: [74602269](https://github.com/Azure/azure-cli/pull/687#discussion_r74602269) [](ancestors = 74602269)